libs/langchain/ragstack_langchain/graph_store/cassandra.py

```diff
@@ -143,7 +127,7 @@ def similarity_search(
 
     def similarity_search_by_vector(
         self, embedding: List[float], k: int = 4, **kwargs: Any
-    ) -> List[Document]:
+    ) -> Iterable[Document]:
```

Comment:
I think we need to keep List as in the signature and change the code below.
I did it in the LangChain PR to be able to satisfy mypy.

Reply:
I don't think this is correct then -- you can't return a `List[Document]` and use `yield` inside the method. `yield` makes it a generator, not a `List`. Specifically, the type annotation here is wrong:

https://github.com/datastax/ragstack-ai/blob/main/libs/langchain/ragstack_langchain/graph_store/cassandra.py#L148

Reply:
Done, but changed to actually return a list.